### DISCUSSION TITLE
What does the monitor metrics mean for milvus？

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#15741] [Q&A and General discussion]
Hi, I am confused for the monitor data of milvus on grafana.
QPM(Query per minute): does it mean nq *(real qps from client)?
Query elapsed time: does it means a search request from client totally cost(search of nq vector), or does it mean a client search request cost = nq * (Query elapsed time) ?

That really matters for me, pls help on this issue.

(I modified qpm to qps)
![image](https://user-images.githubusercontent.com/74544828/155512861-13fc0771-2ab4-432b-af48-b12bfedb4694.png)

![image](https://user-images.githubusercontent.com/74544828/155513066-ce3e93a6-601f-4122-ba81-530125256ecc.png)

**Comment by yhmo** [comment]
QPM(Query per minute) means how many "nq" were executed in a minute. Assume there are 3 search requests finished in 2 minutes, the nq of the 3 requests is 5, 10, 3 respectively. Then the QPM = (5 + 10 +3)/2 = 9

The "Query elapsed time" is the average time cost for each nq. Assume a search request with nq=5, it cost 1 second, the "Query elapsed time"= 1/5=0.2second.
The average cost of nq=5 and nq=1 is different. For each search request, the time consuming includes the following aspects:
(1). decode search parameters by RPC handler
(2). parameters validation
(3). ANN search by CPU
(4). reduce results
(5). encode search results return to client

Typically the (3) is the major cost. If nq=1 cost 10ms in (3), then nq=5 will cost 50ms in (3). But other aspects are almost the same no matter nq=5 or nq=1.
So, you will see the average cost of nq=5 is a bit less than nq=1.

**Reply by xiaojuan8** [reply]
OK, thanks.
